### DISCUSSION TITLE
Warning for Browsers with JIT disabled

### DISCUSSION THREAD

**Discussion by Masgalor** [lldap/lldap#635] [General]
So, I know next to nothing about JIT and WASM and similar frontend technologies.
But I learned that LLDAP's frontend only shows a blank page if it is loaded in a browser that has JIT disabled.

I do understand that the frontend won't work without that feature enabled, but it really took me some time to understand why some of my users only see a blank page. Would it be possible to check if the browser has support for all needed features and show a warning or error if it hasn't? Like if you have javascript disabled completely you will often see a message like "This site needs javascript to work, please enable it."

**Comment by nitnelave** [comment]
Quick note: it's when JavaScript is disabled, not JIT.

We can add a fallback note for users with no JS. Can you open an issue (or convert this into an issue)?

**Reply by Masgalor** [reply]
My wording may be incorrect because I don't understand enough about the topic but I think it is not as simple as that.

Most users reporting this problem to me were using `Vanadium the default GrapheneOS browser`.
It has JavaScript enabled by default so the `<noscript>` html tag did not do anything (maybe I just put it in the wrong place), but to make it work the user had to go to security settings and check `Allow JavaScript JIT`.

I will create an issue for this, no problem.
